iwyu_include_picker.cc

```diff
@@ -1324,18 +1324,50 @@ string IncludePicker::MaybeGetIncludeNameAsWritten(
   return value ? *value : "";
 }
 
-vector<string> IncludePicker::GetCandidateHeadersForSymbol(
-    const string& symbol) const {
-  CHECK_(has_called_finalize_added_include_lines_ && "Must finalize includes");
-  vector<MappedInclude> mapped_includes =
-    GetPublicValues(symbol_include_map_, symbol);
+vector<string> IncludePicker::RelativiseMappedIncludesForPath(
+    const vector<MappedInclude>& includes,
+    const string& including_filepath) const {
+  // Convert each MappedInclude to a quoted include.  Each contains
+  // a default quoted include but sometimes we can do better.
+  // For each path, the iwyu-preprocessor may have stored the quoted-include
+  // as written in including_filepath.  This is better to use than
+  // ConvertToQuotedInclude because it avoids trouble when the same
+  // file is accessible via different include search-paths, or is
+  // accessed via a symlink.
+  // We also try to recalculate any quoted includes using absolute paths,
+  // because they can likely be converted to an include relative to the
+  // current file.
+  const string including_path =
+      MakeAbsolutePath(GetParentPath(including_filepath));
   vector<string> retval;
-  for (const MappedInclude& mapped_include : mapped_includes) {
-    retval.push_back(mapped_include.quoted_include);
+  for (const MappedInclude& mapped_include : includes) {
+    const string& quoted_include_as_written =
+      MaybeGetIncludeNameAsWritten(including_filepath, mapped_include.path);
+    if (!quoted_include_as_written.empty()) {
+      retval.push_back(quoted_include_as_written);
+    } else if (StartsWith(mapped_include.quoted_include, "\"/") &&
+               !mapped_include.path.empty()) {
+      retval.push_back(ConvertToQuotedInclude(mapped_include.path,
+                                              including_path));
+    } else {
+      retval.push_back(mapped_include.quoted_include);
+    }
   }
   return retval;
 }
 
+vector<IncludePicker::MappedInclude>
```

Comment:
This wrapping looks nicer to me, but I think it's counter to convention with the return type on its own line. Do you have clang-format installed, and could you use it on the modified regions (not the whole file!)?

Reply:
Tweaked.  I haven't run clang-format.  Unless you know a clever trick to apply it only to unmodified regions, that seems unwieldy.